### PR TITLE
feat(User): refactor User backend

### PR DESCRIPTION
**Pull request by cldylan**
## Summary

This PR introduces several enhancements to the `User` model and `UserRepository` to support more flexible user data management and improve Firestore update efficiency. The changes were motivated by the need to allow partial updates to user profiles, track pinned trips and images, and reduce unnecessary write operations. The design ensures only specified fields are updated, guest users are protected from modifications, and Firestore documents are safely handled.
It improves flexibility and efficiency of user data updates while extending the `User` model to include pinned trips and images.

#### 1. Add `updateUser` function to `UserRepository`
- Introduces a new suspend function `updateUser` in the `UserRepository` interface.
- Supports optional parameters for `name`, `biography`, `profilePicUrl`, `preferences`, `pinnedTripsUids`, and `pinnedImagesUris` to allow partial updates.
- `UserRepositoryFirebase` implements this by building a map of non-null fields and sending a single `update` request to Firestore for efficiency.
- Includes safeguards:
  - Prevents updates for guest users.
  - Ensures the user document exists before updating.

#### 2. Add pinned images to `User` model
- Adds a new `pinnedImagesUris` field (`List<Uri>`) to the `User` data class.
- `UserRepositoryFirebase` now serializes and deserializes this field to and from Firestore.
- Tests updated to reflect changes to the `User` model.

#### 3. Add pinned trips to `User` model
- Adds a new `pinnedTripsUids` property to the `User` data class.
- Firestore integration updated:
  - Includes `pinnedTripsUids` during user creation and retrieval.
  - Parses `pinnedTripsUids` from Firestore documents.

## Tests

- Added emulator tests for `updateUser` in `UserRepositoryEmulatorTest.kt`.
- Covers:
  - Full updates
  - Partial updates
  - Edge cases (e.g., non-existent users)
- Updated `parseUserFromDoc` to correctly parse `pinnedImagesUris` and `pinnedTripsUids`.

### CODE REVIEW

File: app/src/androidTest/java/com/github/swent/swisstravel/model/user/UserRepositoryEmulatorTest.kt [hunk 1]
```diff
@@ -426,6 +428,115 @@ class UserRepositoryEmulatorTest : InMemorySwissTravelTest() {
...
+        uid = guestUid,
+        name = "Should Not Update",
+        biography = "Guest update",
+        profilePicUrl = "http://example.com/guest.png",
+        preferences = listOf(Preference.SCENIC_VIEWS),
+        pinnedTripsUids = listOf("tripX"),
+        pinnedImagesUris = listOf("file://imageX".toUri()))
+
+    // Nothing to assert in Firestore — test passes if no exception thrown
+    assertTrue(true)
```
Comment (lhegetsc) [inline app/src/androidTest/java/com/github/swent/swisstravel/model/user/UserRepositoryEmulatorTest.kt]: ## Change

Instead of asserting nothing, you should assert that the document does not exist in firestore and that the update didn't create one.

**Review by lhegetsc — CHANGES_REQUESTED** [review]
Overall, good PR. Thanks for these changes! It just needs a few fixes.
